### PR TITLE
[Android] CameraRoll support for Videos

### PR DESCRIPTION
**Pull request by kesha-antonov**
## Motivation

[Android] CameraRoll does not support videos. This PR adds ability to get videos too.

## Test Plan

I don't know how to test it automatically. I even wasn't able to build source code and ran just ```./gradlew :ReactAndroid:installArchives && cp ./android ...``` everytime I changed the code.

And I didn't find any tests for CameraRoll so can't provide any tests. I'm a beginner in java dev.

Please guide me in this.

-----
P.S.
This is a fork from this PR: https://github.com/facebook/react-native/pull/11877
See some details there

### CODE REVIEW

**Comment by kesha-antonov** [comment]
Also I've noticed that vertical videos do have wrong dimensions.
Neither ```MediaStore.Video.VideoColumns.RESOLUTION``` has correct resolution. Android thinks that these videos are horizontal.

My workaround for now is to listen ```onLoad``` event in ```react-native-video``` and get correct dimensions from there. This PR https://github.com/react-native-community/react-native-video/pull/239 added video dimensions

By wrong dimensions I mean: ```640x320``` should be ```320x640```

Maybe you guys can tell me how to get correct dimensions straight from android CameraRoll?

**Comment by zr0n** [comment]
Hey @kesha-antonov , nice beard.
I'm using the plugin react-native-camera-roll-picker (https://www.npmjs.com/package/react-native-camera-roll-picker)
And I tried to merge your changes into my react-native, but when I try to use assetType="Videos" it keep showing only photos.
Am I missing something?

**Comment by kesha-antonov** [comment]
Hey! @zr0n Thanks!

Did you build android after merging?

**Comment by zr0n** [comment]
@kesha-antonov Yes., twice. But when I open the picker, keep showing only photos. Even if I use the assetType="All"  it does not show video files.

**Comment by kesha-antonov** [comment]
I'd suggest to pass some other strings "somestring" as "assetType"
If you'll get error - then you built it successfully.
If not - you should build it with these changes.

Telling this because it's working for me in simulator and a real device.

**Comment by zr0n** [comment]
@kesha-antonov 
Hey Kesha, thank you for answering. 
So, I tried to put some crazy value at assetType and it threw me a red screen with an error. But when I set  assetType to 'Videos' it does not throw error and it shows only images from my gallery.

Any Idea of what is going wrong?

**Comment by amitaymolko** [comment]
@kesha-antonov Whats the status of this PR?

File: ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java [hunk 1]
```diff
@@ -289,69 +324,70 @@ protected void doInBackgroundGuarded(Void... params) {
       // setting a limit at all), but it works because this specific ContentProvider is backed by
       // an SQLite DB and forwards parameters to it without doing any parsing / validation.
       try {
-        Cursor photos = resolver.query(
-            Images.Media.EXTERNAL_CONTENT_URI,
+        Cursor medias = resolver.query(
```
Comment (piotrek1543) [inline ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java]: change to `media`. `Medias` is a wrong phrase.

File: ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java [hunk 2]
```diff
@@ -222,43 +231,48 @@ public void getPhotos(final ReadableMap params, final Promise promise) {
...
     private final Context mContext;
     private final int mFirst;
     private final @Nullable String mAfter;
     private final @Nullable String mGroupName;
+    private final String mAssetType;
     private final @Nullable ReadableArray mMimeTypes;
     private final Promise mPromise;
 
-    private GetPhotosTask(
+    private GetMediasTask(
```
Comment (piotrek1543) [inline ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java]: Change to `GetMediaTask`. I'm pretty sure that one `medium`, multiple `medias`. Check: https://english.stackexchange.com/questions/314099/should-i-use-media-or-medias

**Comment by ManikSmartdata** [comment]
I am using RN version 0.45.0. But the support of Video is only available in RN 0.47.2. I can't update by RN to 0.47.2. It will effect my other node_modules.

Please suggest

File: ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java (hunk 2 above)
Comment (kesha-antonov) [reply ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java]: I think here we get ```medias``` - multiple. Isn't it? Previously we get ```photos```. What's wrong here?

File: ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java (hunk 1 above)
Comment (kesha-antonov) [reply ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java]: Same. Why ```media```? We get multiple items

**Comment by kesha-antonov** [comment]
@hramos Hello! Any update? Any plans to merge? Can you suggest somebody to review?

**Comment by kesha-antonov** [comment]
@zr0n One guy said that he changed his video from Full HD to SD and he was able to see videos. Maybe that's the case. I'm not sure though why I don't have this problem

**Comment by rimbosUY** [comment]
Hi! 
I am very interested to have it solved.  We need to show videos at Android platform.
Please merge the changes to include support.

File: ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java (hunk 2 above)
Comment (Green-Cat) [reply ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java]: Media is already plural, the singular is medium.

Comment (kesha-antonov) [reply ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java]: Ok, Got it

**Comment by hramos** [comment]
Hey, I missed the notifications here. Looks like a merge with master added a bunch of unrelated commits. If you fix this, in a new PR, maybe we can continue working on getting this merged.

**Comment by kesha-antonov** [comment]
@hramos 
Hello!
Yes, I messed up things.
New PR is here: https://github.com/facebook/react-native/pull/16429

Waiting for your feedback!

**Comment by WemagineGuillaume** [comment]
Hi,
I test your new PR but when I try to retrieve list of videos, I have the error : 

> E/unknown:ReactNative: Could not get width/height for content://media/external/images/media/546
                                                               java.io.FileNotFoundException: No entry for content://media/external/images/media/546
                                                                   at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:144)
                                                                   at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:692)
                                                                   at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1158)
                                                                   at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:995)
                                                                   at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:918)
                                                                   at com.facebook.react.modules.camera.CameraRollManager.putImageInfo(CameraRollManager.java:413)
                                                                   at com.facebook.react.modules.camera.CameraRollManager.putEdges(CameraRollManager.java:365)
                                                                   at com.facebook.react.modules.camera.CameraRollManager.access$300(CameraRollManager.java:61)
                                                                   at com.facebook.react.modules.camera.CameraRollManager$GetPhotosTask.doInBackgroundGuarded(CameraRollManager.java:317)
                                                                   at com.facebook.react.modules.camera.CameraRollManager$GetPhotosTask.doInBackgroundGuarded(CameraRollManager.java:249)
                                                                   at com.facebook.react.bridge.GuardedAsyncTask.doInBackground(GuardedAsyncTask.java:34)
                                                                   at com.facebook.react.bridge.GuardedAsyncTask.doInBackground(GuardedAsyncTask.java:22)
                                                                   at android.os.AsyncTask$2.call(AsyncTask.java:304)
                                                                   at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                                   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                                   at java.lang.Thread.run(Thread.java:761)

How can I fix it ?
Thanks

**Comment by kesha-antonov** [comment]
Hi @WemagineGuillaume 

Try this https://github.com/facebook/react-native/pull/16429

I do not support this anymore

**Pull request closed without merging**